Assets/Scripts/CharacterBehaviour.cs

```diff
@@ -0,0 +1,75 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class CharacterBehaviour : MonoBehaviour
+{
+    //MOUSE CONTROL
+    public float lookSpeed, lookXLimit;
+    float rotationX = 0;
+
+    //MOVEMENT PROPERTIES
+    CharacterController characterController;
+    public float walkingSpeed = 5;
+    public float runningSpeed, jumpForce;
+    public bool canMove = true;
+    Vector3 moveDirection = Vector3.zero;
+
+    //PRIVATE ESSENTIALS
+    Camera playerCamera;
+    
+    
+    void Start()
+    {
+        Cursor.lockState = CursorLockMode.Locked;
+
+        characterController = GetComponent<CharacterController>();
+        playerCamera = FindObjectOfType<Camera>().GetComponent<Camera>();
+
+        canMove = true;
+    }
+
+    void Update()
+    {
+        // We are grounded, so recalculate move direction based on axes
+        Vector3 forward = transform.TransformDirection(Vector3.forward);
+        Vector3 right = transform.TransformDirection(Vector3.right);
+        // Press Left Shift to run
+        bool isRunning = Input.GetKey(KeyCode.LeftShift);
+        float curSpeedX = canMove ? (isRunning ? runningSpeed : walkingSpeed) * Input.GetAxis("Vertical") : 0;
+        float curSpeedY = canMove ? (isRunning ? runningSpeed : walkingSpeed) * Input.GetAxis("Horizontal") : 0;
+        float movementDirectionY = moveDirection.y;
+        moveDirection = (forward * curSpeedX) + (right * curSpeedY);
+
+        if (Input.GetButton("Jump") && canMove && characterController.isGrounded)
+        {
+            moveDirection.y = jumpForce;
+        }
+        else
+        {
+            moveDirection.y = movementDirectionY;
+        }
+
+        // Apply gravity. Gravity is multiplied by deltaTime twice (once here, and once below
+        // when the moveDirection is multiplied by deltaTime). This is because gravity should be applied
+        // as an acceleration (ms^-2)
```

Comment:
good comment!

Reply:
This is actually copy-pastad. I take no credit